packages/aws-cdk-lib/aws-s3/README.md

```diff
@@ -622,6 +622,8 @@ as it does not contain any objects.
 To override this and force all objects to get deleted during bucket deletion,
 enable the`autoDeleteObjects` option.
 
+The auto-deletion for objects within S3 buckets is handled by a custom resource. Enabling this feature by setting it to true would additionally include the imposition of the `s3:PutBucketPolicy` to restrict external write access during cleanup, thus mitigating potential race conditions.
```

Comment:
Let's just let this be a continuation of the above text block rather than having it be a standalone. When you do this, we should provide more context about why this being handled by a custom resource is relevant to the policy being added. Right now you mention it being handled by a custom resource but quickly move away from that point. A reader would probably ask themselves how this being handled by a custom resource is related to the `s3:PutBucketPolicy` action. Also, your second sentence can just read, "Enabling this feature would include...". The "setting it true" is already implied by the feature being enabled.

We need to make sure we explain why a `s3:PutBucketPolicy` is being added since this isn't entirely clear upfront especially give that the PR this originated from claims that it, "Adds a DENY policy for S3:PutObject on buckets to be auto-deleted to prevent a race condition on emptying with external bucket writers". This is confusing because `s3:PutBucketPolicy` is not `s3:PutObject` AND `s3:PutBucketPolicy` was added to a policy document with an `Allow` effect, not a `Deny`. I would start with this to understand why `s3:PutBucketPolicy` was added here - https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketPolicy.html